research/conditional/utils/conditional_trainer.py

```diff
@@ -136,6 +140,31 @@ def _after_step_operations(self, step):
         self.model.forward_pass_cache.clear()
         self.layer_manager.manage_learnable_temperature(step)
 
+    def _final_eval(
+        self,
+        n_steps: int,
+    ):
+        if self.current_step == n_steps:
+            self.model.eval()
+            self.gradient_accumulation_steps = 1
+            losses = []
+            for _ in range(self.n_final_eval_batches):
+                batch = self.final_eval_dataloader.get_batch()
+                with torch.no_grad():
+                    loss, _ = self.calculate_loss_and_gradient(
+                        batch, self.batch_size // self.gradient_accumulation_steps
```

Comment:
I don't really get why we divide here by self.gradient_accumulation_steps, while it's also hardcoded to be one. Why does it have to be one as well?